scipy/stats/_bootstrap.py

```diff
@@ -435,8 +450,8 @@ def bootstrap(data, statistic, *, vectorized=True, paired=False, axis=0,
             resampled_data.append(resample)
 
         # Compute bootstrap distribution of statistic
-        theta_hat_b.append(statistic(*resampled_data, axis=-1))
-    theta_hat_b = np.concatenate(theta_hat_b, axis=-1)
+        bootman.update(statistic(*resampled_data, axis=-1))
+    theta_hat_b = bootman.finish()
```

Comment:
@mdhaber I am starting to get back into this work and wanted to ask you some clarifying questions to help me calibrate myself.  Here are my questions (not in any specific order):
* It looks like you introduced `BootstrapManager`, and it currently encapsulates the work previously done by the inline code for computing `theta_hat_b` in batches (one `batch_nominal` samples at a time).
* Currently, `n_resamples` must be specified in order to instantiate `BootstrapManager` (i.e., `n_resamples` cannot be `None`).
* The goal would be to embellish `BootstrapManager` so as to compute the optimal `n_resamples` value automatically.

If this thinking is correct, then I think that I can try to take the next steps (via incremental pull requests into the present pull request).  How to handle `batch_nominal` for the algorithm will need to be figured out in due course.

Thank you.

Reply:
I think those things are right, but I'll try to clarify.
> Currently, `n_resamples` must be specified in order to instantiate `BootstrapManager` (i.e., `n_resamples` cannot be None). The goal would be to embellish BootstrapManager so as to compute the optimal n_resamples value automatically.

Right, my `BootstrapManager` example requires `n_resamples`, but maybe it would help to think of what needs to be done differently. This `BootstrapManager` has a few jobs to do: it keeps track of the number of resamples `k` that have been processed and it stores the bootstrap distribution. You don't have to think about it as improving this implementation of a `BootstrapManager`; you just need to write a class that does the same things.

This might be held up by gh-14334. If we want to use multiprocessing, the way this is structured could be pretty different.